### PR TITLE
activate current language when clicking view_on_site button in admin …

### PR DESCRIPTION
**Pull request by zoe-stockholm**
The issue is in single post admin page, when the post has multi-language, the view_on_site button always navigate to the default language no matter what language the obj is using. The solution is simple to activate the current language and create a url with current language code instead of default language always.

### CODE REVIEW

**Comment by timgraham** [comment]
Thanks for the patch. A couple of notes:
* Please file a Trac ticket for the issue.
* Please remove the ".idea" files added by mistake.
* See our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist) for some helpful tips. In particular, a regression test is needed for the issue.

File: django/contrib/admin/options.py [hunk 1]
```diff
@@ -268,10 +270,14 @@ def get_view_on_site_url(self, obj=None):
             return self.view_on_site(obj)
         elif self.view_on_site and hasattr(obj, 'get_absolute_url'):
             # use the ContentType lookup if view_on_site is True
-            return reverse('admin:view_on_site', kwargs={
+            current_lang = get_language()
+            activate(obj.language_code)
```
Comment (timgraham) [inline django/contrib/admin/options.py]: You can use the `with translation.override(obj.language_code)` context manager to simplify it.

**Comment by zoe-stockholm** [comment]
thanks for the notes. Now I got this error when I tried to check/create ticket tracker. see attached img. Then if I tried to loggin, got 502 Bad Gateway. The server is down or I did something wrong? Appreciate if you can give me some hits or help. Thanks!
error page

**Comment by timgraham** [comment]
Not sure. Could you try again now and give the exact steps you're following to reproduce the error if it still happens?

**Comment by zoe-stockholm** [comment]
Still same error.

I followed this link to "claiming tickets" https://docs.djangoproject.com/en/1.7/internals/contributing/writing-code/submitting-patches/#patch-review-checklist

First, I already created an account to be able to use ticket system. BUT something wrong happened and I got this error during the next step when I tried to click "ticket tracker".

**Comment by timgraham** [comment]
You're using "DjangoProject Login" then? Still not sure of the exact steps to reproduce the problem as this login works for me.

**Comment by timgraham** [comment]
Also, you could try GitHub login.

**Comment by zoe-stockholm** [comment]
sorry, kind of busy last week. but i will try again today and see what's gonna happen.

**Comment by timgraham** [comment]
I just upgrade Trac to 1.0.8. The previous version had a bug: "the session for an authenticated username containing non-alphanumeric characters could not be retrieved, resulting in the user being denied access to every realm and resource." I hope it's fixed now.

**Comment by zoe-stockholm** [comment]
Now the login working! Already newed a ticket and followed the steps.

**Comment by zoe-stockholm** [comment]
Fixed # 25336

**Comment by timgraham** [comment]
After looking at this a bit more, we can't assume that all objects have a `language_code` field (there are test failures with your patch because of this). Can you put this logic in your model's `get_absolute_url()` method instead?

**Comment by zoe-stockholm** [comment]
what if I do:

url = reverse('admin:view_on_site', kwargs={
            'content_type_id': get_content_type_for_model(obj).pk,
            'object_id': obj.pk
        })
if obj.language_code:
    with override(obj.language_code):
        return url
else:
    return url

**Comment by zoe-stockholm** [comment]
url = reverse('admin:view_on_site', kwargs={
'content_type_id': get_content_type_for_model(obj).pk,
'object_id': obj.pk
})
if obj.language_code:
with override(obj.language_code):
return url
else:
return url

**Comment by zoe-stockholm** [comment]
sorry for the Indentation

**Comment by timgraham** [comment]
Are you suggesting that for Django or for your own code? I still haven't entirely understood the issue and how it might be fixed, but I'd suggest to create a sample project that shows the issue and then use [our support channels](https://code.djangoproject.com/wiki/TicketClosingReasons/UseSupportChannels) to ask for help. If it turns out a fix to Django is required, we can reopen the issue.

**Pull request closed without merging**